vector-im/element-android
# FTUE - Msisdn (phone number) entry

Draft as this depends on #6091 

## Type of change


## Content

Partly addresses #6043 - the confirmation screen is currently being designed and will be raised in a separate PR

Adds an updated styling version of the Phone number (Msisdn) entry 

- Extracts the phone number parsing to its own class with tests
- Moves the msisdn 401 as success logic to the wizard delegate with test (matching the email flow)

## Motivation and context

To update the phone number entry flow 

## Screenshots / GIFs

| | | | |
| --- | --- | --- | --- | 
![Screenshot_20220520_140735](https://user-images.githubusercontent.com/1848238/169551050-178b108c-91a7-45b6-97b3-0d634fda271a.png)|![Screenshot_20220520_140744](https://user-images.githubusercontent.com/1848238/169551047-a085e64e-d48a-4121-8498-d071cb85859f.png)|![Screenshot_20220520_140812](https://user-images.githubusercontent.com/1848238/169551041-5cc4cf13-0e1c-4d16-8952-a2d86906bbdd.png)|![Screenshot_20220520_140756](https://user-images.githubusercontent.com/1848238/169551046-d53f211b-e321-4971-8d75-700da134f3f6.png)


## Tests

- Setup a local synase with ThreePid msisdn's enabled 
- Create an account
- See phone number entry screen

## Tested devices

- OS version(s): 31
